lib/docusign/ouath2_strategy.ex

```diff
@@ -0,0 +1,105 @@
+defmodule DocuSign.OAuth2Strategy do
+  @moduledoc ~S"""
+  This module implements an oauth2 strategy for DocuSign.
+
+  ### Examples
+
+  client = DocuSign.OAuth2Strategy.get_token!
+  {:ok, user_info } = OAuth2.Client.get(client, "/oauth/userinfo")
+
+  """
+  use OAuth2.Strategy
+
+  alias OAuth2.{Client, AccessToken, Error}
+  @type param         :: binary | %{binary => param} | [param]
+  @type params        :: %{binary => param} | Keyword.t
+  @type headers       :: [{binary, binary}]
+
+  @private_key Application.get_env(:docusign, :private_key)
+  @token_expires_in Application.get_env(:docusign, :token_expires_in)
+  @hostname Application.get_env(:docusign, :hostname)
+  @client_id Application.get_env(:docusign, :client_id)
+  @user_id Application.get_env(:docusign, :user_id)
+
+  @grant_type "urn:ietf:params:oauth:grant-type:jwt-bearer"
+
+  @doc """
+  Retrieve access token and return a client
+  """
+  @spec get_token!(params, headers, Keyword.t()) :: Client.t() | Error.t()
+  def get_token!(params \\ [], headers \\ [], opts \\ []) do
+    client()
+    |> Client.get_token!(params, headers, opts)
+  end
+
+  @doc """
+  Check expiries of token.
+  return true if token is expired
+  """
+  @spec token_expired?(AccessToken.t()|nil|Client.t()) :: boolean
```

Comment:
It might be a little better to call this `token_needs_refresh?` because it's `true` when the token is `nil`, just just when it's actually expired.